station-worldgen-api-v0/src/main/java/net/modificationstation/stationapi/api/worldgen/surface/SurfaceBuilder.java

```diff
@@ -0,0 +1,157 @@
+package net.modificationstation.stationapi.api.worldgen.surface;
+
+import net.minecraft.block.BlockBase;
+import net.modificationstation.stationapi.api.block.BlockState;
+import net.modificationstation.stationapi.api.tag.TagKey;
+import net.modificationstation.stationapi.api.util.math.Direction.AxisDirection;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.BlockSurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.DepthSurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.HeightSurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.SlopeSurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.StateSurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.SurfaceCondition;
+import net.modificationstation.stationapi.api.worldgen.surface.condition.TagSurfaceCondition;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SurfaceBuilder {
+	private static final ThreadLocal<SurfaceBuilder> INSTANCES = new ThreadLocal<>();
+	private final List<ConditionInfo> conditions = new ArrayList<>();
+	private SurfaceRule rule;
+	
+	private SurfaceBuilder() {}
+	
+	/**
+	 * Start surface building with any starting rule
+	 */
+	public static SurfaceBuilder start(SurfaceRule rule) {
+		SurfaceBuilder instance = INSTANCES.get();
+		
+		if (instance == null) {
+			instance = new SurfaceBuilder();
+			INSTANCES.set(instance);
```

Comment:
Can use `ThreadLocal.withInitial(SurfaceBuilder::new)` instead of a manual check.

Reply:
Fixed